code/datums/mind.dm

```diff
@@ -1562,14 +1562,28 @@
 /datum/mind/proc/find_syndicate_uplink()
 	var/list/L = current.get_contents()
 	for(var/obj/item/I in L)
-		if(I.hidden_uplink)
+		if(I.hidden_uplink) //find items disguised as uplinks
 			return I.hidden_uplink
-	return null
+		if(istype(I, /obj/item/implant/uplink)) //find uplink implants
+			return I
+	return
 
 /datum/mind/proc/take_uplink()
-	var/obj/item/uplink/hidden/H = find_syndicate_uplink()
+	var/list/A = current.actions //remove uplink implant action button
+	for(var/datum/action/item_action/hands_free/activate/I in A)
+		var/obj/item/implant/uplink/U = I.target
+		if(!isnull(U.hidden_uplink))
+			I.Remove(current)
+
+	var/list/L = current.get_contents() //set all hidden_uplink vars to null
+	for(var/obj/item/I in L)
+		if(!isnull(I.hidden_uplink))
+			I.hidden_uplink = null
+
+	var/obj/item/uplink/hidden/H = find_syndicate_uplink() //remove all uplinks
```

Comment:
this comment is incorrect, `find_syndicate_uplink()` will return a single object, if there are multiple `/obj/item/uplink/hidden` in `current` then onl 1 will be QDEL'd and the rest will remain.

Reply:
This made me realize that I missed something very critical in this code. Namely that it only nulls `hidden_uplink` without actually removing the uplinks.
I'm afraid the only way to tackle this is to completely rewrite how `find_syndicate_uplink()` works and make some changes to other procs, which I've done.
This, however, means the PR requires a brand new review from a scratch. In fact, it might even possibly require re-labeling or re-approving.